Packs/VersaDirector/Integrations/VersaDirector/VersaDirector.py

```diff
@@ -0,0 +1,3800 @@
+
+    :rtype: Tuple(boolean,str)
+    """
+
+    try:
+        test_module(client)  # send HTTP request to check if token is valid
+    except DemistoException as e:
+        if e.res.status_code == 401 or "invalid_token" in str(e.message):
+            if current_context:
+                # obtain refresh token and send it using client.access_token_request function
+                refresh_token = current_context.get("refresh_token", "")
+
+                client._headers = None
+
+                # response contains updated auth_token to be used as Bearer token
+                response = client.refresh_token_request(client_id, client_secret, refresh_token)
+                auth_token = response.get("access_token")
+
+                # integration context is updated with new access token and refresh token
+                response["client_id"] = client_id
+                response["client_secret"] = client_secret
+                demisto.setIntegrationContext(response)
+
+                return auth_token
+            else:
+                raise DemistoException("Auth Token Expired, Refresh Token is not found. please create a new Auth Token.")
+
+    return None
+
+
+def create_client_header(
+    use_token: bool,
+    username: str,
+    password: str,
+    client_id: str,
+    client_secret: str,
+    access_token: str,
+):
+    """
+    Creates Auth and Header arguments for the Client object based on the authentication method selected in the integration instance configuration.
+
+    :type use_token: ``bool``
+    :param use_token: boolean argument indicating which type of authentication will be used.
+
+    :type username: ``str``
+    :param username: Username parameter from instance configuration.
+
+    :type password: ``str``
+    :param password: Password parameter from instance configuration.
+
+    :type client_id: ``str``
+    :param client_id: Client ID parameter from instance configuration.
+
+    :type client_secret: ``str``
+    :param client_secret: Client Secret parameter from instance configuration.
+
+    :type access_token: ``str``
+    :param access_token: Access Token parameter from instance configuration.
+
+    :rtype: Tuple(Optional[None,str],dict[str,str])
+    returns (auth, header) tuple
+    """
+    # Basic authentication
+    if not use_token and username and password:
+        credentials = f"{username}:{password}"
+        auth_header = f"Basic {b64_encode(credentials)}"
+        return (username, password), {"Authorization": auth_header}
+    # Auth Token authentication
+    elif use_token and all([client_id, client_secret, access_token]):
+        return None, {"Authorization": f"Bearer {access_token}"}
+    else:
+        raise DemistoException(message="Not all fields for the selected authenticating are set,therefor it cannot be executed.")
+
+
+# HEADING: """ COMMAND FUNCTIONS """
+
+
+def create_auth_client(client: Client, args: Dict[str, Any]) -> CommandResults:
```

Comment:
I would like us to document this auth flow that this function is using. Maybe even add a code comment to check confluence for more info.